ci/travis/install-dependencies.sh

```diff
@@ -1,122 +1,169 @@
+      ;;
+  esac
+  { local set_x="${-//[^x]/}"; } 2> /dev/null  # save set -x to suppress noise
+  set +x
+  local source_line='PYTHON3_BIN_PATH=python; PATH="$HOME/miniconda/bin:$PATH"; export PYTHON3_BIN_PATH PATH;'
+  test -f ~/.bashrc && grep -x -q -F "${source_line}" -- ~/.bashrc || echo "${source_line}" >> ~/.bashrc
+  test -z "${set_x}" || set -x  # restore set -x
+  eval "${source_line}"
+  python -m pip install --upgrade --quiet pip
+}
+
+install_nvm() {
+  if [ "${OSTYPE}" = msys ]; then
+      local version="1.1.7" NVM_HOME="${HOME}/.nvm"
+      if [ ! -f "${NVM_HOME}/nvm.sh" ]; then
+        mkdir -p -- "${NVM_HOME}"
+        export NVM_SYMLINK="${PROGRAMFILES}\nodejs"
+        (
+          cd "${NVM_HOME}"
+          local target="./nvm-${version}.zip"
+          curl -s -L -o "${target}" "https://github.com/coreybutler/nvm-windows/releases/download/${version}/nvm-noinstall.zip"
+          unzip -q -- "${target}"
+          rm -f -- "${target}"
+          printf "%s\r\n" "root: $(cygpath -w -- "${NVM_HOME}")" "path: ${NVM_SYMLINK}" > settings.txt
+        )
+        printf "%s\n" "export NVM_HOME=\"$(cygpath -w -- "${NVM_HOME}")\"" 'nvm() { "${NVM_HOME}/nvm.exe" "$@"; }' > "${NVM_HOME}/nvm.sh"
+      fi
+  fi
+}
+
+install_pip() {
+  local python=python
+  if command -v python3 > /dev/null; then
+    python=python3
+  fi
+
+  if "${python}" -m pip --version || "${python}" -m ensurepip; then  # If pip is present, configure it
+    "${python}" -m pip install --upgrade --quiet pip
+
+    # If we're in a CI environment, do some configuration
+    if [ "${TRAVIS-}" = true ] || [ -n "${GITHUB_WORKFLOW-}" ]; then
+      "${python}" -W ignore -m pip config -q --user set global.disable-pip-version-check True
+      "${python}" -W ignore -m pip config -q --user set global.no-color True
+      "${python}" -W ignore -m pip config -q --user set global.progress_bar off
+      "${python}" -W ignore -m pip config -q --user set global.quiet True
+    fi
+  fi
+}
+
+install_node() {
+  if [ "${OSTYPE}" = msys ]; then
+    { echo "WARNING: Skipping running Node.js due to module incompatibilities with Windows"; } 2> /dev/null
+  else
+    # Install the latest version of Node.js in order to build the dashboard.
+    (
+      set +x # suppress set -x since it'll get very noisy here
+      . "${HOME}/.nvm/nvm.sh"
+      nvm install node
+      nvm use --silent node
+      npm config set loglevel warn  # make NPM quieter
+    )
+  fi
+}
+
+run_npm() {
+  npm ci
+}
+
+install_dependencies() {
+
+  install_base
+  if [ -n "${GITHUB_WORKFLOW-}" ]; then  # Keep Travis's built-in compilers and only use this for GitHub Actions (for now)
+    . "${ROOT_DIR}"/install-toolchains.sh
+  fi
+  install_nvm
+  install_pip
+
+  if [ -n "${PYTHON-}" ]; then
+    install_miniconda
+    pip_packages=(scipy tensorflow=="${TF_VERSION-2.0.0b1}" cython==0.29.0 gym opencv-python-headless pyyaml \
```

Comment:
is TF_VERSION-2.0.0b1 still set?

Reply:
It's the same behavior as before, where it's (still) set only in the Travis YML file. The previous code here was just

    tf_version="$TF_VERSION"
    if [[ $tf_version == "" ]]; then tf_version="2.0.0b1"; fi

which this is equivalent to. Later on when migrating the tests to GitHub Actions we'll have to set the environment variables somehow, but this PR doesn't do that yet.

Reply:
but `TF_VERSION-2.0.0b1` is not set anywhere now?

Reply:
Oh, that's not the variable name :) the notation `${TF_VERSION-2.0.0b1}` means "`${TF_VERSION}` if `TF_VERSION` is set, otherwise `2.0.0b1`. `TF_VERSION` is (still) set in `.travis.yml` like before; I haven't touched it. So it's (almost) equivalent to what the old code was doing. (I could write `${TF_VERSION:-2.0.0b1}` to make it "is set and non-blank" to make it exactly equivalent but this seemed fine too.)